plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/BaseProjectImpl.java

```diff
@@ -588,41 +588,36 @@ protected IStatus run(DBRProgressMonitor monitor) {
                     // Nothing to save and metadata file doesn't exist
                     return Status.OK_STATUS;
                 }
-                try {
-                    if (!CommonUtils.isEmpty(resourceProperties)) {
-                        try (Writer mdWriter = Files.newBufferedWriter(mdFile, StandardCharsets.UTF_8)) {
-                            try (JsonWriter jsonWriter = METADATA_GSON.newJsonWriter(mdWriter)) {
-                                jsonWriter.beginObject();
-
-                                jsonWriter.name("resources");
-                                jsonWriter.beginObject();
-                                for (Map.Entry<String, Map<String, Object>> resEntry : resourceProperties.entrySet()) {
-                                    jsonWriter.name(resEntry.getKey());
-                                    jsonWriter.beginObject();
-                                    Map<String, Object> resProps = resEntry.getValue();
-                                    for (Map.Entry<String, Object> propEntry : resProps.entrySet()) {
-                                        jsonWriter.name(propEntry.getKey());
-                                        Object value = propEntry.getValue();
-                                        if (value == null) {
-                                            jsonWriter.nullValue();
-                                        } else if (value instanceof Number) {
-                                            jsonWriter.value((Number) value);
-                                        } else if (value instanceof Boolean) {
-                                            jsonWriter.value((Boolean) value);
-                                        } else {
-                                            jsonWriter.value(CommonUtils.toString(value));
-                                        }
-                                    }
-                                    jsonWriter.endObject();
-                                }
-                                jsonWriter.endObject();
-
-                                jsonWriter.endObject();
-                                jsonWriter.flush();
+                try (Writer mdWriter = Files.newBufferedWriter(mdFile, StandardCharsets.UTF_8);
+                     JsonWriter jsonWriter = METADATA_GSON.newJsonWriter(mdWriter)
+                ) {
```

Comment:
You don't use `mdWriter` anywhere, so it's redundant.

Additionally, you don't have to call it `jsonWriter`, it's too mouthful and specific in this case, so call it `writer` instead.

```suggestion
                try (JsonWriter writer = METADATA_GSON.newJsonWriter(Files.newBufferedWriter(mdFile, StandardCharsets.UTF_8))) {
```